source/Engine.cpp

```diff
@@ -2343,7 +2347,10 @@ void Engine::DoCollection(Flotsam &flotsam)
 {
 	// Check if any ship can pick up this flotsam. Cloaked ships without "cloaked pickup" cannot act.
 	Ship *collector = nullptr;
-	for(Body *body : shipCollisions.Circle(flotsam.Position(), 5.))
+	vector<Body *> pickupShips;
+	pickupShips.reserve(ships.size());
```

Comment:
Picking up flotsam is typically not done over very long distances, so the amount of ships nearby is unlikely to be larger than 16. Shall we just reserve a size of 16?

Reply:
```suggestion
	pickupShips.reserve(16);
```